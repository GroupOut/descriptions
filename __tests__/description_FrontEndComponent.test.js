// import React from 'react';
// import ReactDOM from 'react-dom';
// import App from './App';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

import React from 'react';
import ReactDOM from 'react-dom';
import Description from '../client/components/description';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from "enzyme-to-json";

// import jsdom from 'jsdom';
// const doc = jsdom.jsdom('<!doctype html><html><body></body></html>')
// global.document = doc
// global.window = doc.defaultView

// console.log(Description);

configure({ adapter: new Adapter() });

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<Description />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

describe('the Description Component', () => {
  it('should render Description', () => {
    const wrapper = shallow(<Description />, { context: {}, disableLifecycleMethods: false })
    console.log(wrapper.debug())
  })

  it('should contain 1 form element', () => {
    const wrapper = shallow(<Description />)
    expect(wrapper.find('form').length).toBe(1)
  })

  it('should contain 1 Description Wrapper', () => {
    const wrapper = shallow(<Description />)
    expect(wrapper.find('.descrip').exists()).toBe(true)
  })

  it('should contain 8 divs', () => {
    const wrapper = shallow(<Description />)
    expect(wrapper.find('div').length).toBe(8)
  })

  it('matches the snapshot', () => {
    const tree = shallow(<Description />)
    expect(toJson(tree)).toMatchSnapshot()
  })

})
