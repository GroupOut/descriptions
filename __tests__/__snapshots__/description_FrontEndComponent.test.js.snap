// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Description Component matches the snapshot 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Choose a deal:
      <input
        onChange={[Function]}
        type="text"
      />
    </label>
    <input
      type="submit"
      value="Submit"
    />
  </form>
  <div
    className="descrip"
  >
    <styled.h3>
      What You'll Get
    </styled.h3>
    <hr />
    <div>
      Kayaking Tour of the Charles River
    </div>
  </div>
  <br />
  <div
    className="addl-info"
  >
    <h3>
      Additional Information
    </h3>
    <div>
      Subject to favorable weather conditions. If canceled due to poor weather, you will be given the option of an alternative date or full refund
    </div>
  </div>
  <br />
  <div
    className="inclusions"
  >
    <h4>
      Inclusions
    </h4>
    <ul>
      <li />
    </ul>
  </div>
  <br />
  <br />
  <div
    className="fine_print"
  >
    <styled.h3>
      The Fine Print
    </styled.h3>
    <hr />
    <div />
  </div>
  <br />
  <div
    className="about"
  >
    <styled.h3>
      About 
    </styled.h3>
    <hr />
    <h5>
      <br />
      <a
        href="https://www.google.com/search?q=undefined "
      >
        Company Website
      </a>
    </h5>
  </div>
  <styled.img
    alt="Store Icon"
    src="./images/store_icon.png"
  />
  <styled.div>
    <styled.h3 />
  </styled.div>
  <br />
  <styled.img
    alt="Map Image"
    src="./images/mapPlaceholder.png"
  />
  <br />
  <styled.div>
    <styled.img
      alt="Map Pin"
      src="/images/pin1.png"
    />
    <styled.div>
      <br />
      <styled.div>
        , 
         
        , 
         
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
